code/modules/mob/mob_movement.dm

```diff
@@ -194,8 +194,10 @@
 
 		if(mob.confused && IsEven(world.time))
 			step(mob, pick(cardinal))
+			mob.last_movement=world.time
 		else
 			. = ..()
+			mob.last_movement=world.time

```

Comment:
I couldn't find a previously defined version of this variable.

Also, good point.